commands/sandbox.go

```diff
@@ -0,0 +1,370 @@
+// with independent 'doctl' commands as needed.
+func Sandbox() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:   "sandbox",
+			Short: "[Beta] Develop functions in a sandbox prior to deploying them in an app",
+			Long: `The ` + "`" + `doctl sandbox` + "`" + ` commands provide a development sandbox for functions.  A sandbox has a local file system component and a cloud component.
+A one-time install of the sandbox software is needed (use ` + "`" + `doctl sandbox install` + "`" + ` to install the software, then ` + "`" + `doctl sandbox connect` + "`" + ` to
+connect to the cloud component of the sandbox provided with your account).  Other ` + "`" + `doctl sandbox` + "`" + ` commands are used to develop and test.`,
+			Aliases: []string{"sbx"},
+			Hidden:  !IsSandboxInstalled(),
+		},
+	}
+
+	// TODO: combine "install" and "connect into a single "enable" command, then "uninstall" should become "disable".
+	// We also need an update strategy.
+	cmdBuilderWithInit(cmd, RunSandboxInstall, "install", "Installs the sandbox support",
+		`This command installs additional software under `+"`"+`doctl`+"`"+` needed to make the other sandbox commands work.
+The install operation is long-running, and a network connection is required.`,
+		Writer, false)
+
+	cmdBuilderWithInit(cmd, RunSandboxUninstall, "uninstall", "Removes the sandbox support", `Removes sandbox support from `+"`"+`doctl`+"`",
+		Writer, false)
+
+	cmdBuilderWithInit(cmd, RunSandboxConnect, "connect <token>", "Connect the cloud portion of your sandbox",
+		`This command connects the cloud portion of your sandbox (needed for testing) by using a provided token.
+You obtain the token from the cloud console (details TBD)`,
+		Writer, false)
+
+	cmdBuilderWithInit(cmd, RunSandboxStatus, "status", "Provide information about your sandbox",
+		`This command reports the status of your sandbox and some details
+concerning its connected cloud portion`, Writer, false)
+
+	cmd.AddCommand(Activations())
+	cmd.AddCommand(Functions())
+	SandboxExtras(cmd)
+	return cmd
+}
+
+// RunSandboxInstall performs the network installation of the 'nim' adjunct to support sandbox development
+func RunSandboxInstall(c *CmdConfig) error {
+	// Check that the sandbox isn't already installed
+	sandboxDir, sandboxExists := getSandboxDirectory()
+	if sandboxExists {
+		return errors.New("An existing sandbox install was detected.  Uninstall before installing again.")
+	}
+	// Make a temporary directory for use during the install
+	tmp, err := ioutil.TempDir("", "doctl-sandbox")
+	if err != nil {
+		return err
+	}
+	// Download the nodejs tarball for this os and architecture
+	goos := runtime.GOOS
+	arch := runtime.GOARCH
+	if arch == "amd64" {
+		arch = "x64"
+	}
+	nodeDir := fmt.Sprintf("node-v%s-%s-%s", NODE_VERSION, goos, arch)
+	URL := fmt.Sprintf("https://nodejs.org/dist/v%s/%s.tar.xz", NODE_VERSION, nodeDir)
+	nodeFileName := filepath.Join(tmp, "node-install.tar.xz")
```

Comment:
Looks like Windows might need some special casing here. `runtime.GOOS` for Windows is `windows` while the Node downloads are just `win`.  Is `tar` available there?

Reply:
There is `tar` for Windows (just about every Unix tool "exists" in that ecosystem).   However, we can't count on it being there.   You are reminding me that there is no shell script for installing `nim` on Windows as there is for other OSs.  We use an installer built using `makensis` to do this ... I think it uses `7zip` to do the unzipping, but I suspect that is bundled into the installer rather than counting on it being on the target machine.   I think we need to get a little Windows-specific expertise on this sub-issue.   Your call on whether we try to fix it in the scope of this PR or make another subtask in SERVERLESS-619 (or wherever you designate).

Reply:
It doesn't necessarily need to happen in this PR as long as we're tracking it elsewhere as we do officially support Windows. Maybe we can just bail out early for now on Windows?

Reply:
I believe the suggestion by @bentranter to stop using `tar` will allow a straightforward fix for this (though not in this PR, there is a follow-on ticket already).   I'll add a fail-fast in this PR if running on windows.